### DISCUSSION TITLE
A few question about validation, train losses and evaluation

### DISCUSSION THREAD

**Discussion by nihil39** [ACEsuit/mace#678] [Q&A] [answered]
Hi, 

I have already asked this in a old thread yesterday but maybe I won't get any answer so I decided to open a new one. 

At the end of one of my training I have this results in the log file 

```
+-------------+---------------------+------------------+-------------------+---------------------------------------+
| config_type | RMSE E / meV / atom | RMSE F / meV / A | relative F RMSE % | RMSE Stress (Virials) / meV / A (A^3) |
+-------------+---------------------+------------------+-------------------+---------------------------------------+
|    train    |         0.2         |       17.4       |        2.30       |                  0.2                  |
|    valid    |         0.2         |       17.6       |        2.32       |                  0.2                  |
|   Default   |         0.2         |       17.6       |        2.34       |                  0.2                  |
+-------------+---------------------+------------------+-------------------+---------------------------------------+
```

They appear to be normal because errors on training is slightly lower than the errors on validation and test sets.

```
{"loss": 0.040587769153017754, "time": 0.7549424171447754, "mode": "opt", "epoch": 2}
{"loss": 0.0038969941308411114, "mae_e": 0.62485660798769, "mae_e_per_atom": 0.0016662842879671736, "rmse_e": 0.7314864436925352, "rmse_e_per_atom": 0.0019506305165134274, "q95_e": 1.30251505414011, ... "time": 2.437013864517212, "mode": "eval", "epoch": 2}
{"loss": 0.037334632390278746, "time": 0.7609601020812988, "mode": "opt", "epoch": 3}
```

I can't understand why in the Results file I have the error on the training set (here I have copied only the last batch but it's the same for other) more or less ten times higher than the error on the validation set (which does not make any sense to me). It is the same for every epoch more or less. 
I set the evaluation interval (`eval_interval`) of my model on every epoch, but what does it mean? Does it mean to compute the loss and other quantities like rmse on energies, forces etc on the validation set? I don't think I have understood this clearly. 

What is the relation between errors and losses? I am using the sress loss function with Energy weight 1, Force weight = Stress weight = 1

**Comment by ilyes319** [comment, marked as answer]
Hello,
This behavior can happen even though it is not common. Usually the cause is the presence of noisy data. The reason why the evaluated model can be better than the trained one is because they are not the same. The evaluated model is an exponential moving average (ema) of the trained model over all its epoch. If your training is quite noisy, it is possible that you see a big different between the performance of non ema model compared to the ema model.

**Reply by HarshitBuilds** [reply]
Hi @ilyes319,
Even I am facing a similar issue as faced by @nihil39 here. After training the MACE model on AIMD data my training loss is close to 10x the validation loss. Is it okay to proceed ahead even then?
![image](https://github.com/user-attachments/assets/83e04f67-8ae1-4679-a686-972add7765bc)

**Reply by ilyes319** [reply]
@HarshitBuilds Can you share your log file so i have a look, your energy errors are unusually high. And yes it is not a problem for the loss functions to be higher in training than validation.

**Reply by HarshitBuilds** [reply]
Hi Ilyes,
Thanks a lot for your prompt reply. I am sharing the log file with you. I am a little new to MACE (MLIPs in general) but have gone through the paper and the tutorials on the MACE website. But please do point out any mistakes if you see in the work. 
[3mo6cl_mace_run-12365.log](https://github.com/user-attachments/files/20726125/3mo6cl_mace_run-12365.log)
Thanks!

**Reply by HarshitBuilds** [reply]
Hi @ilyes319 
Could you please help me with my high errors for energy obtained from the model. Tuning the hyperparameters also doesn't change it much. 

One possibility I am considering is that the issue lies in the AIMD data itself which was used to train the model. To investigate this, I plotted a histogram representing the energy distribution in the 1000 training frames (see below), to find any possible issues. 

![image](https://github.com/user-attachments/assets/6b01dfae-b94f-4e5a-8341-423a26a9b529)

I am sharing the AIMD data as well.
[3mo6cl_1000_frames](https://drive.google.com/file/d/1wI0r80mTj8QmRe3-6t2qegWI91kvaBbc/view?usp=sharing)
Thanks

**Reply by ilyes319** [reply]
Can you do the same plot, but plot the interaction energy per atom (substract the sum of E0s and then divide per number of atoms)?

**Reply by HarshitBuilds** [reply]
Yes, of course. Here's the plot for atomization energy per atom for my 3Mo6Cl cluster.

![image](https://github.com/user-attachments/assets/5ae77f86-7c16-4b45-9e19-ca286ab7a3af)

**Reply by ilyes319** [reply]
How did you compute your data? Did you make sure to put the forces as the -gradient? There seems to be an inconsistency between forces and energies, so might a sign problem or a unit problem. Do check the forces are in eV/A.

**Reply by HarshitBuilds** [reply]
Hi Ilyes,
Sorry for the late reply. We did find some issues in the AIMD data that we used for training the MACE model. The AIMD data was generated using VASP software (version: 6.4.1). We made some changes in the parameters for running the AIMD simulation like: 
Temperature reduction from 2000K to 800K.
Reducing timestep from 1fs to 0.5fs.
Adding LDIPOL correction 

The result was that we got much improvement in the metrics for the same system as shown in the figure.
![image](https://github.com/user-attachments/assets/217b00e6-32d1-472c-8ddd-e2ba101a78c8)
Thanks!

**Comment by gabor1** [comment]
Is it possible that you took the potential energy rather than the free energy from VASP? Always use the free energy, that is what is consistent with the forces.
